kubetest/anywhere_test.go

```diff
@@ -54,16 +57,48 @@ func TestNewKubernetesAnywhere(t *testing.T) {
 			name:              "kubeadm with specific versions",
 			phase2:            "kubeadm",
 			kubeadmVersion:    "unstable",
+			kubeletVersion:    "foo",
 			kubernetesVersion: "latest-1.6",
 			expectConfigLines: []string{
 				".phase2.provider=\"kubeadm\"",
 				".phase2.kubeadm.version=\"unstable\"",
 				".phase2.kubernetes_version=\"latest-1.6\"",
+				".phase2.kubelet_version=\"foo\"",
+				".phase3.weave_net=y",
+			},
+		},
+		{
+			name:              "kubeadm with ci kubelet version",
+			phase2:            "kubeadm",
+			kubeadmVersion:    "unstable",
+			kubeletCIVersion:  "vfoo",
+			kubernetesVersion: "latest-1.6",
+			expectConfigLines: []string{
+				".phase2.provider=\"kubeadm\"",
+				".phase2.kubeadm.version=\"unstable\"",
+				".phase2.kubernetes_version=\"latest-1.6\"",
+				".phase2.kubelet_version=\"gs://kubernetes-release-dev/ci/vfoo/bin/linux/amd64/\"",
+				".phase3.weave_net=y",
+			},
+		},
+		{
+			name:              "kubeadm with ci kubelet version",
+			phase2:            "kubeadm",
+			kubeadmVersion:    "unstable",
+			kubeletCIVersion:  "latest",
+			kubernetesVersion: "latest-1.6",
+			expectConfigLines: []string{
+				".phase2.provider=\"kubeadm\"",
+				".phase2.kubeadm.version=\"unstable\"",
+				".phase2.kubernetes_version=\"latest-1.6\"",
+				".phase2.kubelet_version=\"gs://kubernetes-release-dev/ci/vbar/bin/linux/amd64/\"",
 				".phase3.weave_net=y",
 			},
 		},
 	}
 
+	// TODO(jessicaochen) figure out how to restore non-test readGSFile behavior without causing tests to hang
```

Comment:
A defer should be enough. Here's an example:

https://github.com/kubernetes/test-infra/blob/ad19d04798049201a82c70639900bba593e740d6/kubetest/extract_test.go#L113-L124